rollup.config.js

```diff
@@ -0,0 +1,58 @@
+import resolveNode from '@rollup/plugin-node-resolve';
+import typescript from '@rollup/plugin-typescript';
+import commonjs from '@rollup/plugin-commonjs';
+import json from '@rollup/plugin-json';
+import { terser } from 'rollup-plugin-terser';
+import externalGlobals from 'rollup-plugin-external-globals';
+import babel from '@rollup/plugin-babel';
+
+const tsOpts = {
+    target: 'es2018',
+    strict: true,
+    noImplicitReturns: true,
+    esModuleInterop: true,
+    jsx: 'react',
+    include: ['src/**/*'],
+    exclude: ['node_modules', '**/*.test.ts', 'src/factories/*', 'src/cdk/*'],
+    tsconfig: false,
+};
+
+const globals = {
+    react: 'automat.react',
+    emotion: 'automat.emotion', // TODO remove this dependency
+    '@emotion/core': 'automat.emotionCore',
+    'emotion-theming': 'automat.emotionTheming',
+};
+
+export default {
+    input: 'src/components/modules/ContributionsEpic.tsx',
+    output: {
+        file: 'dist/modules/Epic.js',
+        format: 'es',
+        sourcemap: false,
+    },
+    external: id => Object.keys(globals).some(key => id.startsWith(key)),
+    plugins: [
+        resolveNode(),
+        commonjs(),
+        json(),
+        babel({
+            extensions: ['.ts', '.tsx', '.js', '.jsx', '.es6', '.es', '.mjs'],
+            plugins: [['emotion', { sourceMap: false }]],
+            presets: [
+                [
+                    '@babel/preset-env',
+                    {
+                        targets: {
+                            ie: '11',
```

Comment:
Definitely an optimisation and something to consider separately, but as modern browsers have considerable support for ES2015 syntax, we could spike the creation of two bundles, one transpiled to ES5 for older browsers, another to ES6 for newer browsers. The service could default to serving the ES5 bundle to be on the safe side, but allowing the request to indicate that a bundle with newer syntax may be acceptable. This could be done by looking at the UA, or adding a custom HTTP header whose value could be set by checking support for a significant ES6 feature. The goal of all this would be to reduce the bundle size by leveraging native browser support for ES6 features. A spike would help assess the cost/benefits of this approach. Happy to get involved any time.

Reply:
Yeah agree, this could definitely be something we optimise later.